### PR TITLE
Add warning about missing doctype

### LINKED ISSUE #2: Move doctype checking to the element mixin
Currenlty, `<vaadin-grid>` checks for doctype declaration at https://github.com/vaadin/vaadin-grid/blob/fb02d64d35342d471b804bd2338ae07ee5a8edb9/src/vaadin-grid.html#L622

Move this functionality to the common vaadin-element-mixin

### PR DESCRIPTION
**Pull request by web-padawan**
Fixes #2

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-element-mixin/16)

### CODE REVIEW

**Review by yuriy-fix — APPROVED** [review]
:lgtm:

> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/vaadin/vaadin-element-mixin/16#-:-LJrbGg2AyUhpSRFKn6J:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved

**Review by tomivirkki — APPROVED** [review]
:lgtm:

> Reviewed 2 of 3 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/vaadin/vaadin-element-mixin/16#-:-LJrf2xFDyXVRihEJg1C:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved

File: test/doctype.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      beforeEach(done => {
+        customElements.whenDefined('x-element').then(done);
+      });
+
+      it('should warn about doctype', () => {
+        const _warn = console.warn;
+        const spy = console.warn = sinon.spy();
+        fixture('default');
+        console.warn = _warn;
+        expect(spy.called).to.be.true;
```
Comment (limonte) [inline test/doctype.html:43]: Would it be better to use `calledWith` (https://sinonjs.org/releases/v2.0.0/assertions/) and verify that the warning message is correct?

E.g. https://github.com/sweetalert2/sweetalert2/blob/13662349768bfdd98cfb91000e55bf2c3cfc3bb9/test/qunit/params/type.js#L11

File: vaadin-element-mixin.html [hunk 2]
```diff
@@ -34,6 +34,14 @@
           window.Vaadin.registrations.push(this);
         }
       }
+      ready() {
+        super.ready();
+        if (document.doctype === null) {
```
Comment (limonte) [inline vaadin-element-mixin.html:39]: According to https://developer.mozilla.org/en-US/docs/Web/API/Document/doctype it should be possible to check the actual doctype:

`if (document.doctype.name !== 'html') {`

File: test/doctype.html (hunk 1 above)
Comment (limonte) [reply test/doctype.html:43]: O-ho, I'm apparently too slow :)

File: vaadin-element-mixin.html (hunk 2 above)
Comment (web-padawan) [reply vaadin-element-mixin.html:39]: I think this statement is enough for us:

> The property returns null if there is no DTD associated with the current document.

See https://developer.mozilla.org/en-US/docs/Web/HTML/Quirks_Mode_and_Standards_Mode

> In almost standards mode, there are only a very small number of quirks implemented.

Checked the "Transitional" DTD and it works:

```
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
```

Comment (limonte) [reply vaadin-element-mixin.html:39]: Fine by me.

FYI, for the "Transitional" DTD `document.doctype.name` is still `'html'`.

**Pull request merged**